librz/arch/isa/mips/il/common.c

```diff
@@ -0,0 +1,1024 @@
+	RzILOpPure *rt = MIPS_REG(1);
+	RzILOpPure *target = MIPS_IMM(2);
+
+	return BRANCH(NE(rs, rt), JMP(target), NOP());
+}
+
+static RzILOpEffect *mips_il_clo(const csh *handle, const cs_insn *insn, const ut32 gprlen) {
+	// Count Leading Ones in Word
+	NOT_IMPLEMENTED;
+}
+
+static RzILOpEffect *mips_il_clz(const csh *handle, const cs_insn *insn, const ut32 gprlen) {
+	// Count Leading Zeros in Word
+	NOT_IMPLEMENTED;
+}
+
+static RzILOpEffect *mips_il_div(const csh *handle, const cs_insn *insn, const ut32 gprlen) {
+	RzILOpPure *rs = MIPS_REG(0);
+	RzILOpPure *rt = MIPS_REG(1);
+
+	RzILOpPure *quotient = SDIV(DUP(rs), DUP(rt));
+	RzILOpPure *remainder = SMOD(rs, rt);
+
+	RzILOpEffect *set_lo = SETG(MIPS_REG_LO, quotient);
+	RzILOpEffect *set_hi = SETG(MIPS_REG_HI, remainder);
+	return SEQ2(set_lo, set_hi);
+}
+
+static RzILOpEffect *mips_il_divu(const csh *handle, const cs_insn *insn, const ut32 gprlen) {
+	RzILOpPure *rs = MIPS_REG(0);
+	RzILOpPure *rt = MIPS_REG(1);
+
+	RzILOpPure *quotient = DIV(DUP(rs), DUP(rt));
+	RzILOpPure *remainder = MOD(rs, rt);
+
+	RzILOpEffect *set_lo = SETG(MIPS_REG_LO, quotient);
+	RzILOpEffect *set_hi = SETG(MIPS_REG_HI, remainder);
+	return SEQ2(set_lo, set_hi);
+}
+
+static RzILOpEffect *mips_il_ext(const csh *handle, const cs_insn *insn, const ut32 gprlen) {
+	MIPS_CHECK_IF_TARGET_IS_ZERO_REG_AND_NOP();
+
+	// Extract Bit Field (EXT rt, rs, pos, size)
+	const char *rt = REG(0);
+	RzILOpPure *rs = MIPS_REG(1);
+	RzILOpPure *pos = MIPS_IMM(2);
+	RzILOpPure *size = MIPS_IMM(3);
+
+	return SETG(rt, EXTRACT32(rs, pos, size));
+}
+
+static RzILOpEffect *mips_il_ins(const csh *handle, const cs_insn *insn, const ut32 gprlen) {
+	MIPS_CHECK_IF_TARGET_IS_ZERO_REG_AND_NOP();
+
+	// Insert Bit Field (INS rt, rs, pos, size)
+	RzILOpPure *rt = MIPS_REG(0);
+	RzILOpPure *rs = MIPS_REG(1);
+	RzILOpPure *pos = MIPS_IMM(2);
+	RzILOpPure *size = MIPS_IMM(3);
+
+	return SETG(REG(0), DEPOSIT32(rt, pos, size, rs));
+}
+
+static RzILOpEffect *mips_il_j(const csh *handle, const cs_insn *insn, const ut32 gprlen) {
+	RzILOpPure *target = MIPS_IMM(0);
+	return JMP(target);
+}
+
+static RzILOpEffect *mips_il_jal(const csh *handle, const cs_insn *insn, const ut32 gprlen) {
+	RzILOpPure *jump_target = MIPS_IMM(0);
+	RzILOpEffect *link_op = MIPS_LINK();
+	RzILOpEffect *jmp_op = JMP(jump_target);
+	return SEQ2(link_op, jmp_op);
+}
+
+static RzILOpEffect *mips_il_jalr(const csh *handle, const cs_insn *insn, const ut32 gprlen) {
+	RzILOpEffect *link_op = NULL;
+	RzILOpPure *jump_target = NULL;
+	if (OPCOUNT() < 2) {
+		link_op = MIPS_LINK();
+		jump_target = MIPS_REG(0);
+	} else {
+		link_op = SETG(REG(0), UN(gprlen, insn->address + 8));
+		jump_target = MIPS_REG(1);
+	}
+
+	RzILOpEffect *jmp_op = JMP(jump_target);
+	return SEQ2(link_op, jmp_op);
+}
+
+static RzILOpEffect *mips_il_jalx(const csh *handle, const cs_insn *insn, const ut32 gprlen) {
+	// Jump and Link Exchange
+	// TODO: handle change of isamode: ISAMode = !ISAMode
```

Comment:
Please don't forget to open an issue about this.

Reply:
no need. this is a cpu bit that needs to be flipped when emulating.

Reply:
when i implement these regs, i will handle it. this is needed also for FPU stuff.